### ISSUE TITLE
Missing required property on the radio component.

### ISSUE THREAD

**Issue by MrHBS** [material-components/material-web#4316]
Missing required property on the checkbox and radio components. Also the switch component could use aria-required (and aria-readonly/disabled?) I think?

**Comment by asyncLiz** [comment]
At the moment IDREF aria properties such as `aria-controls` or `aria-labelledby` aren't well supported with shadow dom, so I don't think those will be easy features.

Adding `required` to checkbox and radio groups along with validity seem like a reasonable feature.

**Comment by asyncLiz** [comment]
Checkbox has required now, radio is still missing it
